src/components/payroll/AdvancedFiltersDialog.js

```diff
@@ -0,0 +1,284 @@
+  getDefaultAppliedCustomFilters,
+  readOnly,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  // eslint-disable-next-line no-unused-vars
+  const [currentFilter, setCurrentFilter] = useState({
+    field: '', filter: '', type: '', value: '', amount: '',
+  });
+  const [filters, setFilters] = useState(getDefaultAppliedCustomFilters());
+
+  const createParams = (moduleName, objectTypeName, uuidOfObject = null) => [
+    `moduleName: "${moduleName}"`,
+    `objectTypeName: "${objectTypeName}"`,
+    uuidOfObject !== null ? `uuidOfObject: "${uuidOfObject}"` : '',
+  ];
+
+  const fetchFilters = (params) => fetchCustomFilter(params);
+
+  const handleOpen = () => {
+    setFilters(getDefaultAppliedCustomFilters());
+    setIsOpen(true);
+  };
+
+  const handleClose = () => {
+    setCurrentFilter(CLEARED_STATE_FILTER);
+    setIsOpen(false);
+  };
+
+  const handleRemoveFilter = () => {
+    setCurrentFilter(CLEARED_STATE_FILTER);
+    setAppliedFiltersRowStructure([CLEARED_STATE_FILTER]);
+    setFilters([CLEARED_STATE_FILTER]);
+  };
+
+  const handleAddFilter = () => {
+    setCurrentFilter(CLEARED_STATE_FILTER);
+    setFilters([...filters, CLEARED_STATE_FILTER]);
+  };
+
+  function updateJsonExt(inputJsonExt, outputFilters) {
+    const existingData = JSON.parse(inputJsonExt || '{}');
+    if (!existingData.hasOwnProperty('advanced_criteria')) {
+      existingData.advanced_criteria = [];
+    }
+    const filterData = JSON.parse(outputFilters);
+    existingData.advanced_criteria = filterData;
+    const updatedJsonExt = JSON.stringify(existingData);
+    return updatedJsonExt;
+  }
+
+  const saveCriteria = () => {
+    setAppliedFiltersRowStructure(filters);
+    const outputFilters = JSON.stringify(
+      filters.map(({
+        filter, value, field, type,
+      }) => ({
+        custom_filter_condition: `${field}__${filter}__${type}=${value}`,
+      })),
+    );
+    const jsonExt = updateJsonExt(objectToSave.jsonExt, outputFilters);
+    updateAttributes(jsonExt);
+    setAppliedCustomFilters(outputFilters);
+    handleClose();
+  };
+
+  useEffect(() => {
+    if (object && isEmptyObject(object) === false) {
+      let paramsToFetchFilters = [];
+      if (objectType === BENEFIT_PLAN) {
+        paramsToFetchFilters = createParams(
+          moduleName,
+          objectType,
+          isBase64Encoded(object.id) ? decodeId(object.id) : object.id,
+        );
+      } else {
+        paramsToFetchFilters = createParams(
+          moduleName,
+          objectType,
+        );
+      }
+      fetchFilters(paramsToFetchFilters);
+    }
+  }, [object]);
+
+  useEffect(() => {}, [filters]);
+
+  return (
+    <>
+      <Button
+        onClick={handleOpen}
+        variant="outlined"
+        color="#DFEDEF"
+        className={classes.button}
+        style={{
+          border: '0px',
+          textAlign: 'right',
+          display: 'block',
+          marginLeft: 'auto',
+          marginRight: 0,
+        }}
```

Comment:
This can be done within scope of new ticket related to moving to core module. At this stage when I move outside of component it doesn't work as expected.